visualize-admin/visualization-tool
# Dashboard filters

Work in progress.

### Functionality

It is now possible to activate a single shared temporal filter for all charts depending on a particular temporal dimensions.

- [x] Shared time slider, takes unit from dimension
- [x] UI to activate/deactivate year filter
- [x] Ability to change default from/to  of the time rane
- [x] Dashboard filter is not shown inside the "configurating" view
- [x] Chart config gains a new attribute dashboardFilters where sharedFilters are stored

### How to test

1. Start from Photovoltaikanlagen
2. Add a line chart
3. Add a new dataset like Median electricity tariff for Switzerland
4. Proceed to layout options
5. Choose Dashboard as a layout
6. See that a section "Shared filters" appears in the left panel
7. Activate the shared filter on Year
8. You can now interactively control both charts with the same filter
9. Control the default extent through the date pickers in the left panel
10. Publish the chart

### Next steps in following PRs

- Implement the logic for data filters
- Implement the logic for "point in time" filter

### Refactors

Take the opportunity to explode the big configurator-state file into multiple ones. With this, it should be easier to add changes. I'd like your feedback on this Bartosz also.
